### PR TITLE
Update vista3d to use MONAI components

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Fixes # .

### Description
A few sentences describing the changes proposed in this pull request.

### Status
**Ready/Work in progress/Hold**

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [ ] In-line docstrings updated.
- [ ] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [ ] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [ ] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [ ] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [ ] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [ ] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

**Review by KumoLiu — COMMENTED** [review]
Thanks for the PR, looks good to me.

Did you try with the model weight to see if it can loaded correctly and the results make sense?

**Comment by yiheng-wang-nv** [comment]
Hi @KumoLiu ,
Yes, I tested with multiple sample images and visualize the results

**Review by KumoLiu — APPROVED** [review]

File: models/vista3d/scripts/evaluator.py [hunk 1]
```diff
@@ -243,8 +243,8 @@ def _iteration(self, engine: SupervisedEvaluator, batchdata: dict[str, torch.Ten
                 label_set = np.arange(output_classes).tolist()
             label_prompt = torch.tensor(label_set).to(engine.state.device).unsqueeze(-1)
             # point prompt is generated withing vista3d,provide empty points
-            points = torch.zeros(label_prompt.shape[0], 1, 3)
-            point_labels = -1 + torch.zeros(label_prompt.shape[0], 1)
+            points = torch.zeros(label_prompt.shape[0], 1, 3).to(inputs.device)
```
Comment (yiheng-wang-nv) [inline models/vista3d/scripts/evaluator.py:246]: HI @heyufan1995 , could you help to confirm if the changes here are correct? I got device mismatch error thus changed here, thanks in advance.

**Pull request merged**